### ISSUE TITLE
Tune cell size factor directly

### ISSUE THREAD

**Issue by FG-TUM** [AutoPas/AutoPas#768]
**Is your feature request related to a problem? Please describe.**

Currently, the cell size factor is part of our configuration, and (all) possible values are tested until the best (estimate) is found. This is inefficient and even impossible if a continuous interval is given.

**Describe the solution you'd like**
Each container should aim for an optimal number of particles per cell and thus can directly calculate its optimal cell size factor. The optimal number of particles could either be given by the user (meh), hardcoded (meh²), or determined automatically via a microbenchmark, either during `AutoPas::init` or via an extra benchmark program and then fed, e.g., via a file.

**Additional context**
This would lay the foundation for the idea of determining hardware parameters independent of the tuning cycle.

**Comment by SamNewcome** [comment]
I agree, some parameters may be better to tune with a microbenchmark and this is fully something we should support. This is similar to what is done in the Kernel Tuner software, which we could use for inspiration for this micro benchmarking.

Do you have evidence that there is an optimal number of particles per cell which holds for some functor on some hardware regardless of the scenario?

**Comment by FG-TUM** [comment]
Yes-ish. This was often benchmarked in some student theses but with all the changes over time this should probably be revisited.

**Comment by FG-TUM** [comment]
[This paper](https://apps.dtic.mil/sti/pdfs/ADA364892.pdf) is rather old, and the hardware they used is probably barely comparable to today's, but their data (which they dumped into the paper as a huge table...) suggests that there is such a thing as an optimal number of particles per cell. The paper is about cells smaller than the cutoff.

**Comment by thesamriel** [comment]
#1156 introduces a microbenchmarking framework for SoA sorting.
Further parameters like CSF could be added on top of it.
